pkg/operation/botanist/etcdencryption.go

```diff
@@ -158,36 +152,21 @@ func confChecksum(conf *apiserverconfigv1.EncryptionConfiguration) (string, erro
 // encryption configuration changed. Rewriting here means that a patch request is sent that forces
 // the etcd to encrypt them with the new configuration.
 func (b *Botanist) RewriteShootSecretsIfEncryptionConfigurationChanged(ctx context.Context) error {
-	checksum := func() string {
-		b.mutex.RLock()
-		defer b.mutex.RUnlock()
-		return b.CheckSums[common.EtcdEncryptionSecretName]
-	}()
-
-	if err := b.rewriteShootSecretsIfChecksumChanged(ctx, checksum); err != nil {
-		return err
-	}
-
-	gardenSecret := &corev1.Secret{ObjectMeta: kutil.ObjectMetaFromKey(common.GardenEtcdEncryptionSecretKey(b.Shoot.Info.Namespace, b.Shoot.Info.Name))}
-	_, err := controllerutil.CreateOrUpdate(ctx, b.K8sGardenClient.Client(), gardenSecret, func() error {
-		kutil.SetMetaDataAnnotation(gardenSecret, common.EtcdEncryptionChecksumAnnotationName, checksum)
-		return nil
-	})
-	return err
-}
-
-func (b *Botanist) rewriteShootSecretsIfChecksumChanged(ctx context.Context, checksum string) error {
-	secret := &corev1.Secret{}
-	if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, common.EtcdEncryptionSecretName), secret); err != nil {
+	gardenerResourceList := gardencorev1alpha1helper.GardenerResourceDataList(b.ShootState.Spec.Gardener)
```

Comment:
Instead of parsing the `b.ShootState`, can't you use `b.Shoot.ETCDEncryption` now?

Reply:
During restoration we can skip the generation and upsert-to-shootstate step so `b.Shoot.ETCDEncryption` will be nil.
If we don't skip those steps then I can take it from `b.Shoot.ETCDEncryption.

Reply:
OK, why not simply also setting `b.Shoot.ETCDEncryption` configuration? We already have this field now, so I'd say it makes sense to only parse the `ShootState` at most once, store the to-be-used encryption configuration in `b.Shoot.ETCDEncryption` and use it from there always?

Reply:
Yep was also thinking of that. Wondering if I should just create a `ShootState` struct similar to the `Shoot` one and keep all relevant information there - `b.ShootState.ETCDEncryptionConfig` and the `ShootState` would just be in `b.ShootState.Info`. Could be useful for the Secrets/Certificates although I'm not sure if I can think of a concrete case right now. Maybe it makes sense to create that struct if necessary with the Certificates PR.

Reply:
Yes, sounds good, let's check it again with the certificate PR. For now I'd keep the structure as is so that we can proceed with this PR.